In-App Marketplace: use `useQuery` instead of `getQuery` for tab URLs, so browser history works

andfinally opened the pull request:
This is a small update to the changes in https://github.com/woocommerce/woocommerce/pull/38885. We're using `useQuery` instead of `getQuery` from `@woocommerce/navigation` to ensure going back and forward in browser history works with the new marketplace page and tabs.

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

- Changes URL query handler in `Tabs` component to use `useQuery` instead of `getQuery`.

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://href.li/?17315-gh-Automattic/woocommerce.com
Related: https://github.com/woocommerce/woocommerce/pull/38885

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

1. Check out this branch.
2. Enable the right Node version:

```
nvm use
```

3. From the WooCommerce plugin folder, update the Composer autoload class maps:

```
cd plugins/woocommerce
composer dump-autoload
```

4.  Then run the build:

```
pnpm run --filter=woocommerce build
```

5. Go to WooCommerce in your local environment, and confirm that you see the new menu item `Marketplace` in the WooCommerce menu above `Settings`. (If you're using wp-env the URL is http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fmarketplace.)
6. Check that you can navigate from there to the `Extensions` tab and back again. The `Marketplace` menu item should be highlighted as the current item while you're on each of these tabs, and the URL should correctly reflect the page and tab. If you visit one of the URLs, the correct tab should load.
7. Check that navigation works as expected when you use the browser Back and Forward buttons.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Message

Tweaks the method we use to get the current query string, to ensure browser history works when you navigate to the marketplace.

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

### Screenshots

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/008c2a3d-c102-4847-86a7-893aaae5023f">

Dan-Q reviewed (approved):
Works a treat. Although I guess we should now change a lot of these "Marketplace" to "Extensions", including in URLs? 😵‍💫

andfinally committed:
Changed `Tabs` to use `useQuery` instead of `getQuery`, so the browser back button will work.
Changed submenu item name back to `Extensions`, and keeping it in the current order within the WooCommerce menu.
Changed `path` value to `extensions`.

kdevnel reviewed (approved):
Looks great and works perfectly! Thanks for this Gareth.

Side note, I'm not sure how necessary the changelog is for every PR. If you untick the "generate changelog automatically" in your description, that requirement isn't checked anymore. I wonder if we can skip this each time and only do a manual changelog after we merge the feature branch 🤔

andfinally commented:
In line with Dan's [comment](https://github.com/woocommerce/woocommerce/pull/39099#pullrequestreview-1534764701), I've restored the name in the menu item from `Marketplace` to `Extensions`, and changed the URLs to `admin.php?page=wc-admin&path=%2Fextensions`. This looks slightly awkward when the tab is also `extensions`, but it's probably OK. I've also changed the priority on the filter that creates the menu item, so the menu item is in the same position it is currently, after `Settings` and `Status`. This is wrong order IMHO, but it's probably better to correct this another time.
